src/alert-syslog.c

```diff
@@ -209,6 +209,13 @@ static TmEcode AlertSyslogIPv4(ThreadVars *tv, const Packet *p, void *data)
     if (p->alerts.cnt == 0)
         return TM_ECODE_OK;
 
+    char proto[16] = "";
+    if (SCProtoNameValid(IPV4_GET_IPPROTO(p))) {
+        strlcpy(proto, known_proto[IPV4_GET_IPPROTO(p)], sizeof(proto));
```

Comment:
this seems like an unnecessary copy. We can use a ptr instead. For the non-SCProtoNameValid we can use a local string and point the same pointer to that